test/functional/rpc_generateblock.py

```diff
@@ -51,37 +88,44 @@ def run_test(self):
         assert_equal(len(block['tx']), 1)
         assert_equal(block['tx'][0]['vout'][0]['scriptPubKey']['addresses'][0], combo_address)
 
-        # Generate 110 blocks to spend
-        node.generatetoaddress(110, address)
+        # Generate some blocks to spend
+        miniwallet.generate(10)
+        node.generate(100)
 
         # Generate some extra mempool transactions to verify they don't get mined
         for _ in range(10):
-            node.sendtoaddress(address, 0.001)
+            miniwallet.send_self_transfer(from_node=node)
 
         self.log.info('Generate block with txid')
-        txid = node.sendtoaddress(address, 1)
+        utxo = utxos.pop(0)
+        rawtx = create_miniwallet_rawtx(utxo=utxo, scriptPubKey=scriptPubKey)
+        txid = node.sendrawtransaction(rawtx)
```

Comment:
I'm not seeing how this is different than `miniwallet.send_self_transfer(...)`

Reply:
You're right, it really isn't. It mostly just comes down to the blocks generated by `generateblock` not being appended to `self._utxos`. So what it comes down to is what the proper approach for appending to `self._utxos` is.